### PR TITLE
Disable Output

### LINKED ISSUE #11: Account for nodes whose outputs are not requested in the MBDyn model
The output of specific nodes can be disabled in MBDyn, for example using the 
```
output, no
```
directive at the end of their definition in the input file. 

Blendyn gathers the information about the nodes present in the model from the .log file, in which no information about their output is given. So it is possible that a node without any output is imported in the Blender scene, causing trouble when the animation is requested. 

Blendyn should be able to recognise that the output of specific nodes has been disabled, and just warn the user without stopping the animation of the other nodes.

### PR DESCRIPTION
**Pull request by janga1997**
I've only worked out a solution for netcdf output. Will push commits for .mov files soon.
Fixes #11

### CODE REVIEW

**Comment by zanoni-mbdyn** [comment]
I've tried out your solution with the model that you find attached to this message. 
After having imported all the nodes and the elements, I tried to animate and Blender crashed with this error and a segfault:

```
Traceback (most recent call last):
  File "/home/zanoni/.config/blender/2.78/scripts/addons/blendyn-master/base.py", line 1245, in invoke
    return self.execute(context)
  File "/home/zanoni/.config/blender/2.78/scripts/addons/blendyn-master/base.py", line 1236, in execute
    ret_val = set_motion_paths_netcdf(context)
  File "/home/zanoni/.config/blender/2.78/scripts/addons/blendyn-master/baselib.py", line 723, in set_motion_paths_netcdf
    first = nc.variables[node_var + 'X'][int(tdx), :]
KeyError: 'node.struct.0.X'
```
you can see the relevant portion of the blender.crash.txt report attached.
Hope this helps debugging. 
[triangle_mass.txt](https://github.com/zanoni-mbdyn/blendyn/files/1077684/triangle_mass.txt)
[blender.crash.txt](https://github.com/zanoni-mbdyn/blendyn/files/1077689/blender.crash.txt)

**Comment by janga1997** [comment]
My bad. I made the temporary assumption that nodes would be numbered from 1 -- Number_of_nodes.

Correcting it now.

**Comment by zanoni-mbdyn** [comment]
Thanks, I'll test tomorrow

**Comment by zanoni-mbdyn** [comment]
Works fine. I don't know if it would be worth to inform the user in the `.bylog` file that some nodes' outputs were not found... What do you guys think?

**Comment by janga1997** [comment]
It seems a bit unnecessary to me, IMO.
If the user is disabling the output of a node, he must be doing it explicitly, as it doesn't happen by accident.

**Comment by zanoni-mbdyn** [comment]
Could be, but he could also be simulating and visualizing a model that was made by someone else. Or he might have forgot to re-enable the output of a ceirtain node, etc..

**Comment by janga1997** [comment]
Ok, but generating a self.report() message is a little tricky.
So here is what we will do, we will log it into the `*.bylog` file, generate a self.report() message, pause for a second to let the user see it, and then move on to see the value of `ret_val`. (See my changes in base.py for context)

**Comment by zanoni-mbdyn** [comment]
That seems a good approach to me.

**Comment by janga1997** [comment]
We can log it into the `.bylog` file, but generating a self.report() message doesn't work unless there's a **return** following it.
I think just logging it will be enough.

File: base.py [hunk 1]
```diff
@@ -784,14 +784,11 @@ def execute(self, context):
         missing = context.scene.mbdyn.missing
         if len(obj_names) > 0:
             message = "Some of the nodes/elements are missing in the new .log file"
-            self.report({'WARNING'}, message)
             baseLogger.warning(message)
             hide_or_delete(obj_names, missing)
-            return {'FINISHED'}
 
         if len(mbs.disabled_output) > 0:
```
Comment (zanoni-mbdyn) [inline base.py]: this is the bare minimum, but I'd like to tell explicitly to the user which nodes do not print their output. Something like:
```
MBDynReadLog(): WARNING: no output found for node 3
MBDynReadLog.execute(): WARNING: no output found for node 5
MBDynReadLog.execute(): WARNING: no output found for node13 
...
```
Also, please keep the name of the reporting method in the log: it helps debugging sometimes.

**Comment by zanoni-mbdyn** [comment]
Sorry I wrote the review several days ago but I forgot to submit it!

**Comment by janga1997** [comment]
There isn't a clean/easy way to get the class/method name of every logging call. What we could do instead is get the filename and line_number of the logging call generated. That should help more in debugging right?
Example:-
```
06/30/2017 04:24:25 PM - WARNING - [base.py:792] - No output for nodes 1
06/30/2017 04:24:25 PM - INFO - [base.py:833] - MBDyn entities imported successfully
06/30/2017 04:25:44 PM - ERROR - [plotlib.py:411] - Please save current Blender file first

```

**Comment by zanoni-mbdyn** [comment]
Yeah, that could work

**Comment by janga1997** [comment]
I already pushed the commit.

**Comment by zanoni-mbdyn** [comment]
I don't know why, but now I don't get any warning or log messages at all with your latest commit. 
Have you tested it? If so, with which MBDyn input?

**Comment by janga1997** [comment]
I tested it with both text and netcdf output, and with saved and untitled blender files.
If you press `Animate Scene`, do your animations proceed as expected? Then it should probably work.
I am looking into a bit more closely.

**Comment by zanoni-mbdyn** [comment]
The animation part is fine, it is just that I don't get any warnings or reports at all

**Comment by janga1997** [comment]
@zanoni-mbdyn You are not going to get a warning/report in Blender, as we discussed the problems above, [comment](https://github.com/zanoni-mbdyn/blendyn/pull/24#issuecomment-310645751),
we can only log it in the `.bylog` file. There should be messages logged in your `bylog` file.

**Comment by zanoni-mbdyn** [comment]
> we can only log it in the .bylog file. There should be messages logged in your bylog file.

That's where I was looking for them, and didn't find any.

File: base.py [hunk 2]
```diff
@@ -773,15 +778,18 @@ class MBDynReadLog(bpy.types.Operator):
...
         if len(obj_names) > 0:
             message = "Some of the nodes/elements are missing in the new .log file"
-            self.report({'WARNING'}, message)
             baseLogger.warning(message)
             hide_or_delete(obj_names, missing)
-            return {'FINISHED'}
+
+        if len(mbs.disabled_output) > 0:
+            message = "No output for nodes " + mbs.disabled_output
+            baseLogger.warning(message)
```
Comment (janga1997) [inline base.py:792]: @zanoni-mbdyn I'm really at a loss for what's happening here.
If your animation works as expected after disabling a node or more, then that means `len(mbs.disabled_output) > 0`.
If that wasn't the case, your animation wouldn't work.
Could just give me a copy of the log file to check?

Comment (zanoni-mbdyn) [reply base.py:792]: My bad, now I see the correct output. I don't exactly understand what i did wrong, but now it works. 

I'm going to merge.

**Comment by zanoni-mbdyn** [comment]
Merged

**Pull request merged**